docs/rtk-query/usage/persistence-and-rehydration.mdx

```diff
@@ -37,28 +37,33 @@ box with the `autoMergeLevel1` or `autoMergeLevel2` [state reconcilers](https://
 when persisting the root reducer, or with the `autoMergeLevel1` reconciler when persisting just the api reducer.
 
 ```ts title="redux-persist rehydration example"
-import type { Action, PayloadAction } from '@reduxjs/toolkit'
+import type { Action } from '@reduxjs/toolkit'
 import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react'
 import { REHYDRATE } from 'redux-persist'
 
 type RootState = any // normally inferred from state
 
-function isHydrateAction(action: Action): action is PayloadAction<RootState> {
+function isHydrateAction(action: Action): action is Action<typeof REHYDRATE> & {
```

Comment:
For me it seems that simply using `any` on the return type of `extractRehydrationInfo` is enough to break the circular reference, and then the type check can be done with
```tsx
function isHydrateAction(action: Action): action is PayloadAction<RootState> {
    return action.type === HYDRATE;
}
```

Reply:
yeah, i was just adjusting the type guard to what the hydrate action actually looks like, to avoid the `action as any` when accessing action.key